### ISSUE TITLE
unmaintained class-validator and nestjs hard dependencies

### ISSUE THREAD

**Issue by armingjazi** [nestjs/nest#7814]
## Is your feature request related to a problem? Please describe.

Currently, nestjs comes with a hard dependency on class-validator and class-transformer, those packages are not being maintained anymore (except bots updating every now and then), the current maintainers are not responding to any messages.

## Describe the solution you'd like

it should be possible to use forked versions of those packages in nestjs, Of course it is possible to define custom validation pipes and then use the forked ones, but personally I think that should be plan-b

## What is the motivation / use case for changing the behavior?

forgive me for the rough write-up, I would really only like to open up the discussion as the discord channel always gets messy and stuff get lost.

**Comment by jmcdo29** [comment]
This is actually something I'd like to work with as well, making a `@nestjs/validation` or something like that that has the specific `ValidationPipe` and `ClassSerializationInterceptor` so that they can be removed from the main `@nestjs/common` package. 

This could also lead up to having better extensibility with swapping out packages like `@deepkit/types`.

**Comment by kamilmysliwiec** [comment]
This has been already discussed in the past https://github.com/nestjs/nest/issues/4553#issuecomment-611355263

TL;DR We might consider forking these libs in case any severe issue pops up (that nobody will want to address on their side). This doesn't seem to be the case atm though.
